src/service.cc

```diff
@@ -1305,19 +1305,17 @@ int service::handle_async_check_result(check_result* queued_check_result) {
           << get_plugin_output();
   }
 
-  /* get the host that this service runs on */
-  temp_host = (host*)this->host_ptr;
-
+  host* hst{get_host_ptr().get()};
```

Comment:
can we use a shared_ptr here directly ? or even use directly get_host_ptr instead of hst ?